src/libraries/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -255,14 +256,37 @@ private static TSource[] HashSetToArray<TSource>(HashSet<TSource> set)
 
         private static List<TSource> HashSetToList<TSource>(HashSet<TSource> set)
         {
-            var result = new List<TSource>(set.Count);
+            int count = set.Count;
 
-            foreach (TSource item in set)
+            List<TSource> result;
+            if (count > 0)
             {
-                result.Add(item);
+                result = new List<TSource>(count);
+                Span<TSource> span = SetCountAndGetSpan(result, count);
+
+                int index = 0;
+                foreach (TSource item in set)
+                {
+                    span[index] = item;
+                    ++index;
+                }
+
+                Debug.Assert(index == span.Length, "All list elements were not initialized.");
+            }
+            else
+            {
+                result = new List<TSource>();
             }
 
             return result;
         }
+
+        internal static List<TSource> ToSingleItemList<TSource>(TSource item)
+        {
+            List<TSource> list = new List<TSource>(1);
+            Span<TSource> span = SetCountAndGetSpan(list, 1);
+            span[0] = item;
+            return list;
```

Comment:
Is this measurable better than just `new List<TSource>(1) { item }`, in which case this method needn't exist at all?

Reply:
Gains seem negligible given we're forcing another check for every other input size. I'd recommend removing this.

Reply:
Removed